pkgs/tools/typesetting/sile/default.nix

```
@@ -42,56 +44,69 @@ let
   ] ++ lib.optionals (lib.versionOlder lua.luaversion "5.3") [
     compat53
   ]);
-in
 
-stdenv.mkDerivation (finalAttrs: {
+  in stdenv.mkDerivation (finalAttrs: {
   pname = "sile";
-  version = "0.14.17";
+  version = "0.15.4";
 
   src = fetchurl {
-    url = "https://github.com/sile-typesetter/sile/releases/download/v${finalAttrs.version}/sile-${finalAttrs.version}.tar.xz";
-    sha256 = "sha256-f4m+3s7au1FoJQrZ3YDAntKJyOiMPQ11bS0dku4GXgQ=";
+    url = "https://github.com/sile-typesetter/sile/releases/download/v${finalAttrs.version}/sile-${finalAttrs.version}.tar.zst";
+    sha256 = "0n13kv1pabqccjycswcfafhks54l9ck6k8dfkd436jggharkgn1m";
```

generally SRI hash is preferred (you could get this hash from e.g., `nix-hash --to-sri --type sha256 0n13kv1pabqccjycswcfafhks54l9ck6k8dfkd436jggharkgn1m`)

`zstd` is moved in here to facilitate `fetchCargoTarball`

```suggestion
    hash = "sha256-Ndg3s4LvSTNIm66haSZLlBQ9oVOOcc28ZAwvdcOeI1g=";
    nativeBuildInputs = [ zstd ];
```

---

Thanks! It looks like the current invocation for that hash conversion is `nix hash convert --hash-algo sha256`. I'm a bit surprised `nix-prefetch-url` doesn't just output the preferred format though by default.

---

fair point, I've now created a feature req here: https://github.com/NixOS/nix/issues/10990